cilium-health/cmd/root.go

```diff
@@ -170,7 +170,7 @@ func runServer() {
 	common.RequireRootPrivilege(targetName)
 
 	// Wait until Cilium API is available
-	waitForCiliumAPI()
+	// waitForCiliumAPI()
```

Comment:
FWIW I believe that the intention behind this wait previously was that if you don't wait for the Cilium API to become available, then cilium-health is more likely to complain in the logs about the lack of ability to talk to Cilium (since it will almost immediately start reaching out to cilium to figure out the list of IPs that it should start polling).

Reply:
Not only. I also noticed that if you start cilium-health server before cilium-agent, you have two servers running:

```
root@runtime1:~# systemctl status cilium-health
● cilium-health.service - cilium-health
   Loaded: loaded (/lib/systemd/system/cilium-health.service; disabled; vendor preset: enabled)
   Active: active (running) since Wed 2018-12-05 20:00:45 UTC; 3min 44s ago
     Docs: http://docs.cilium.io
 Main PID: 14230 (cilium-health)
    Tasks: 10 (limit: 4684)
   CGroup: /system.slice/cilium-health.service
           └─14230 /usr/bin/cilium-health --daemon

Dec 05 20:00:45 runtime1 systemd[1]: Started cilium-health.
Dec 05 20:00:47 runtime1 cilium-health[14230]: level=info msg="Serving cilium health at http://[::]:4240" subsys=health-server
Dec 05 20:00:49 runtime1 cilium-health[14230]: level=info msg="Serving cilium health at unix:///var/run/cilium/health.sock" subsys=health-server
root@runtime1:~# systemctl cat cilium-health
```

When you uncomment that line, you have only one server running, the one which listenes on U nix socket. That seems strange, I'm trying to debug deeper...

Anyway, I think we should not wait for Cilium API. I guess that waiting for Cilium API was the cause of timeouts in CI.